### PR TITLE
Proper use of pagination in AWS queries

### PR DESCRIPTION
**Pull request by uberblah**
To permit more than a single page of results in any resource listing.

### CODE REVIEW

File: lib/terraforming/resource/kms_key.rb [hunk 1]
```diff
@@ -44,20 +44,47 @@ def tfstate
...
-          .reject { |key| managed_master_key?(key) }
-          .map { |key| @client.describe_key(key_id: key.key_id) }
-          .map(&:key_metadata)
+        marker = ""
+        keys = []
+
+        loop do
+          resp = @client.list_keys(marker: marker)
+          keys += resp
+                    .keys
```
Comment (dtan4) [inline lib/terraforming/resource/kms_key.rb]: __[rubocop]__
* Align `.keys` with `resp` on line 66.

File: lib/terraforming/resource/kms_key.rb [hunk 2]
```diff
@@ -44,20 +44,47 @@ def tfstate
...
+                    .reject { |key| managed_master_key?(key) }
```
Comment (dtan4) [inline lib/terraforming/resource/kms_key.rb]: __[rubocop]__
* Align `.reject` with `resp` on line 66.

File: lib/terraforming/resource/kms_key.rb [hunk 3]
```diff
@@ -44,20 +44,47 @@ def tfstate
...
+                    .map { |key| @client.describe_key(key_id: key.key_id) }
```
Comment (dtan4) [inline lib/terraforming/resource/kms_key.rb]: __[rubocop]__
* Align `.map` with `resp` on line 66.

File: lib/terraforming/resource/kms_key.rb [hunk 4]
```diff
@@ -44,20 +44,47 @@ def tfstate
...
+                    .map(&:key_metadata)
```
Comment (dtan4) [inline lib/terraforming/resource/kms_key.rb]: __[rubocop]__
* Align `.map` with `resp` on line 66.

File: lib/terraforming/resource/db_security_group.rb [hunk 5]
```diff
@@ -46,7 +46,17 @@ def ingresses_of(security_group)
       end
 
       def db_security_groups
-        @client.describe_db_security_groups.map(&:db_security_groups).flatten.select { |sg| !ingresses_of(sg).empty? }
+        marker = ""
+        dbsgs = []
+
+        loop do
+          resp = @client.describe_db_security_groups(marker: marker)
+          dbsgs += resp.db_security_groups.select { |sg| !ingresses_of(sg).empty? }
```
Comment (dtan4) [inline lib/terraforming/resource/db_security_group.rb]: __[rubocop]__
* Use `reject` instead of inverting `select`.

**Comment by dtan4** [comment]
Sorry for late review 🙇 

As described at https://aws.amazon.com/jp/blogs/developer/response-paging/ , paging loop can be written more simply using Enumerator. I prefer this style.

```ruby
iam = Aws::IAM.new
iam.list_users.each do |response|
  puts response.users.map(&:user_name)
end
```

File: lib/terraforming/resource/auto_scaling_group.rb [hunk 6]
```diff
@@ -67,7 +67,17 @@ def tfstate
       private
 
       def auto_scaling_groups
-        @client.describe_auto_scaling_groups.map(&:auto_scaling_groups).flatten
+        token = ""
```
Comment (dtan4) [inline lib/terraforming/resource/auto_scaling_group.rb]: I think we don't have to rewrite this code...

As described at https://aws.amazon.com/jp/blogs/developer/response-paging/ , pagination can be written more simply using loop.

```ruby
@client.describe_auto_scaling_groups.map(&:auto_scaling_groups).flatten
```

can be rewritten as:

```
asgs = []
@client.describe_auto_scaling_groups.each do |response|
  asgs << response.auto_scaling_groups
end.flatten
```

Comment (uberblah) [inline lib/terraforming/resource/auto_scaling_group.rb]: Ah, sorry about that then.

**Pull request closed without merging**